python/lsst/sims/catUtils/mixins/PhotometryMixin.py

```diff
@@ -620,3 +621,133 @@ def get_lsst_magnitudes(self):
+                magList = bandpassDict.magListForSed(dummySed)
+                self._ssmMagDict[magTag] = magList
+                self._ssmMagNormDict[sedName] = magNorm
+            else:
+                dmag = magNorm - self._ssmMagNormDict[sedName]
+                magList = self._ssmMagDict[magTag] + dmag
+            magListOut.append(magList)
+
+        return numpy.array(magListOut).transpose()
+
+
+    @compound('lsst_u','lsst_g','lsst_r','lsst_i','lsst_z','lsst_y')
+    def get_lsst_magnitudes(self):
+        """
+        getter for LSST magnitudes of solar system objects
+        """
+
+        if not hasattr(self, 'lsstBandpassDict'):
+            self.lsstBandpassDict = BandpassDict.loadTotalBandpassesFromFiles()
+
+        return self._magnitudeGetter(self.lsstBandpassDict)
+
+
+    @compound('dmagTrailing', 'dmagDetection')
+    def get_ssm_dmag(self):
+        """
+        This getter will calculate:
+
+        dmagTrailing: the offset in m5 used to represent the loss in signal to noise
+        resulting from the fact that the object's motion smears out its PSF
+
+        dmagDetection: the offset in m5 used to represent the shift in detection
+        threshold resulting from the fact that the object's motion smears out
+        its PSF
+        """
+
+        if self.obs_metadata.seeing is None:
+            raise RuntimeError("Cannot calculate dmagTraling/dmagDetection. "
+                               "Your catalog's ObservationMetaData does not "
+                               "specify seeing.")
+
+        if not isinstance(self.obs_metadata.bandpass, str):
+            raise RuntimeError("dmagTrailing/dmagDetection calculation is confused. "
+                               "Your catalog's ObservationMetaData contains multiple "
+                               "bandpasses.  It is unclear what seeing value you mean. "
+                               "Re-create your catalog with only one bandpass.")
+
+        if not hasattr(self, 'photParams') or self.photParams is None:
+            raise RuntimeError("You cannot calculate dmagTrailing/dmagDetection. "
+                               "Your catalog does not have an associated PhotometricParameters "
+                               "member variable.  It is impossible to know what the exposure time is.")
+
+        dradt = self.column_by_name('velRa') # in radians per day (actual sky velocity;
+                                             # i.e., no need to divide by cos(dec))
+
+        ddecdt = self.column_by_name('velDec') # in radians per day
+
+        if len(dradt)==0:
+            return numpy.zeros((2,0))
+
+        a_trail = 0.76
+        b_trail = 1.16
+        a_det = 0.42
+        b_det = 0.00
+        seeing = self.obs_metadata.seeing[self.obs_metadata.bandpass] # this will be in arcsec
+        texp = self.photParams.nexp*self.photParams.exptime
+        velocity = numpy.sqrt(numpy.power(numpy.degrees(dradt),2) + numpy.power(numpy.degrees(ddecdt),2))
```

Comment:
off topic: I want to get the velocity (rather than just separate velRa/velDec) into the output catalogs, so that it's accessible for MAF. I need to add another mixin method? (and can I put it into this PR?)